apps/system/lockscreen/js/lockscreen.js

```diff
@@ -851,15 +852,15 @@
    */
   LockScreen.prototype.checkPassCode =
   function lockscreen_checkPassCode(passcode) {
-    var request = {
-      passcode: passcode,
-      onsuccess: this.onPasscodeValidationSuccess.bind(this),
-      onerror: this.onPasscodeValidationFailed.bind(this),
-    };
-    window.dispatchEvent(new CustomEvent(
-      'lockscreen-request-passcode-validate',
-      { detail: request }
-    ));
+    PasscodeHelper.check(passcode).then((result) => {
```

Comment:
Yes, sorry.

The goal is to have onls one passcode and all access to it (e.g., testing and setting the passcode) should go through the PasscodeHelper. This way we can improve the security of all apps that use the lockscreen by just improving the PasscodeHelper (e.g. cryptographic hash strength).

Does that answer your question?